src/Synapse/Synapse/Common/AzureSessionCredential.cs

```diff
@@ -51,31 +51,36 @@ public AzureSessionCredential(IAzureContext DefaultContext, DebugLogWriter logWr
 
         public override AccessToken GetToken(TokenRequestContext requestContext, CancellationToken cancellationToken)
         {
-            AccessToken token;
+            DateTimeOffset expiresOn;
+            string token = string.Empty;
             accessToken.AuthorizeRequest((tokenType, tokenValue) =>
             {
-                token = new AccessToken(tokenValue, DateTimeOffset.UtcNow);
+                token = tokenValue;
+                expiresOn = DateTimeOffset.UtcNow;
```

Comment:
I know it's not introduced by your change, but does it make sense for "expiresOn" to be "now"?

Reply:
> I know it's not introduced by your change, but does it make sense for "expiresOn" to be "now"?

@isra-fel  we can follow up in another issue https://github.com/Azure/azure-powershell/issues/25368